packages/default-theme/src/logic/useLocales.js

```diff
@@ -0,0 +1,36 @@
+import { computed } from "@vue/composition-api"
+import languagesMap from "sw-plugins/languages"
+import { getApplicationContext } from "@shopware-pwa/composables"
+
+export const useLocales = (rootContext) => {
+  const { i18n, router } = getApplicationContext(rootContext, "useLocales")
+
+  // TODO: consider using availabeDomains config to list the languages
+  const availableLanguages = computed(() => Object.values(languagesMap) || [])
+  // TODO: consider using locale code from current domains config
+  const currentLocale = computed(() => i18n.locale)
+
+  const changeLocale = async (localeCode) => {
+    // look for current domain
+    const domainForLocaleFound = rootContext.$routing.availableDomains.find(
```

Comment:
same here - use values from getApplicationContext instead of rootContext directly